thoth/adviser/sieves/abi_compat.py

```diff
@@ -59,31 +59,38 @@ def pre_run(self):
                             python_version=runtime_environment.python_version
                         ))
 
+        self.cache = {}
+
         _LOGGER.debug("Analyzed image has the following symbols: %r", self.image_symbols)
 
     def run(self, package_versions: Generator[PackageVersion, None, None]):
         """If package requires non-present symbols remove it."""
-        for package_version in package_versions:
-            package_symbols = self.context.graph.get_python_package_required_symbols(
-                                package_name=package_version.name,
-                                package_version=package_version.locked_version,
-                                index_url=package_version.index.url)
+        for pkg_vers in package_versions:
+
+            cache_key = f"{pkg_vers.name}-{pkg_vers.locked_version}-{pkg_vers.index_url}"
+            if cache_key not in self.cache:
+                self.cache[cache_key] = set(self.context.graph.get_python_package_required_symbols(
```

Comment:
This seemed to bring the sieve up to a reasonable speed

Reply:
Nice! This is the way to go to speed things up :+1:. 

We have a cache layer in thoth-storage - you can configure it there:

https://github.com/thoth-station/storages/blob/974cff49c78df36d4d1d7f12b22cf686ee98240f/thoth/storages/graph/postgres.py#L616-L617

And register it in:

https://github.com/thoth-station/storages/blob/974cff49c78df36d4d1d7f12b22cf686ee98240f/thoth/storages/graph/postgres.py#L4724